Add postgresql13 documentation (#733)
uyuni-project/uyuni-docs: modules/upgrade/pages/db-migration-13.adoc

.Procedure: Preparing to Upgrade

. Check the active PostgreSQL version:
+
----
psql --version
----
+
If you are using PostgreSQL{nbsp}12, you can upgrade to PostgreSQL{nbsp}13.
+
If you are already using PostgreSQL version 13, you do not need to perform this migration.
. Check the active smdba version:
+
----
rpm -q smdba
----
+
// CHECKIT!
+
PostgreSQL{nbsp}13 requires ``smdba`` version 1.7.6 or later.

. Perform a database backup.
    For more information on backing up, see xref:administration:backup-restore.adoc[].

[[db-migration-13-upgrade]]
== Upgrade PostgreSQL

[WARNING]
====
Always create a database backup before performing a migration.
====

PostgreSQL upgrades can be performed in two ways: a regular upgrade, or a fast upgrade:

A regular upgrade will create a complete copy of the database, so you will need double the existing database size of space available.
Regular upgrades can take a considerable amount of time, depending on the size of the database and the speed of the storage system.

A fast upgrade only takes a few minutes, and uses almost no additional disk space.
However, if a fast upgrade fails, you must restore the database from the backup.

> A fast upgrade reduces the risk of running out of disk space.

Reviewer: ... but increase the risk of data lose when a backup does not exist or cannot be replayed

+
// CHECKIT! ====>>>====>>>=======================================> ^^^^

The [path]``pg-migrate-12-to-13.sh`` script performs these operations:

* Stop spacewalk services
* Shut down the running database
* Check if PostgreSQL{nbsp}13 is installed and install it if necessary
* Switch from PostgreSQL{nbsp}12 to PostgreSQL{nbsp}13 as the new default
* Initiate the database migration
* Create a PostgreSQL configuration file tuned for use by {productname}
* Start the database and spacewalk services

[NOTE]
====
If the upgrade fails, the migration script will attempt to restore the database to its original state.
====

.Procedure: Performing a Fast PostgreSQL Upgrade
. Perform a database backup.
    Without a verified database backup, you must not initiate a fast upgrade.
    For more information on backing up, see xref:administration:backup-restore.adoc[].
. Start the upgrade:
+
----
/usr/lib/susemanager/bin/pg-migrate-12-to-13.sh fast
----
. When the upgrade has successfully completed, you can safely delete the old database directory and reclaim lost disk space.
    The old directory is renamed to [path]``/var/lib/pgsql/data-pg12``.

> // CHECKIT! ====>>>====>>>=======================================> ^^^^

Reviewer: @jordimassaguerpla will also the "fast" upgrade create such a directory? I thought it is more an implace upgrade. Would you check this please?

Author: fixed

Author: Yes it will create the directory. The "fast upgrade" only adds the "--links" option to the pg_update command, but the directory is created anyway by the migration script.

[[db-migration-13]]
= Database Migration from Version 12 to 13

This section covers upgrading the PostgreSQL database from version{nbsp}12 to version{nbsp}13.
If you are already using PostgreSQL 13, you do not need to perform this migration.
If you are using version 10, see xref:upgrade:db-migration-12.adoc[].
If you are using an older version, such as version 9.6, see xref:upgrade:db-migration-10.adoc[].

> If you want to upgrade to the latest {productname} version, you must be using PostgreSQL version 12 or 13, depending on the underlying operating system. When running on SLES 15 SP3, version 13 should be used. When running on Leap 15.2, version 12 should be used.

Reviewer 2: ```suggestion
If you want to upgrade to the latest {productname} version, you must be using PostgreSQL version 12 or 13, depending on the underlying operating system.

If you are running SLES 15 SP3, use PostgreSQL 13. 
If you are running Leap 15.2, use PostgreSQL 12.
```

////
== Background Information

* {productname} 3.1.4, SLES 12 SP3, Postgres 9.6
* {productname} 3.1.11, SLES 12 SP3, Postgres 9.6
* {productname} 3.2.10, SLES 12 SP3, Postgres 9.6
* {productname} 3.2.10, SLES 12 SP3, Postgres 10
* {productname} 3.2.10, SLES 12 SP4, Postgres 10
* {productname} 4.0.0, SLES 15 SP1, Postgres 10
* {productname} 4.1.0, SLES 15 SP2, Postgres 12
* {productname} 4.2.0, SLES 15 SP3, Postgres 13

> 

Reviewer 2: I feel like this is actually useful information for the reader. Can we turn it into a table perhaps?

Author: I am not that familiar with asciidoc and actually I don't know how to render it to test it.

How would a table look like in asciidoc?

Author: Are we talking about this?

```
[options="header"]
|=================================================
| SUSE Manager version | SLES version | Postgresql version
| {productname} 3.1.4 | SLES 12 SP3 | Postgres 9.6                                
| {productname} 3.1.11 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 10                                
| {productname} 3.2.10 | SLES 12 SP4 | Postgres 10                                
| {productname} 4.0.0 | SLES 15 SP1 | Postgres 10                                 
| {productname} 4.1.0 | SLES 15 SP2 | Postgres 12                                 
| {productname} 4.2.0 | SLES 15 SP3 | Postgres 13
|=================================================
```

Reviewer 2: You can look at the install guide requirements chapter, the client cfg guide, or the reference guide to see examples, but this is the basic syntax:

```
[[ref-id]]
.Title
[cols="1,1,1", options="header"]
|===
| Column 1                  | Column 2                 | Column 3
| Content in column 1 | Content in column 2 | Content in column 3
|=== 
```

All good if you don't want to do this, though, just raise a new issue and I'll sort it out :)

Author: Should the ref-id be "ref-id" or be some kind of id? If so, which?

I have this now

`
////                                                                            
== Background Information                                                       
[[ref-id]]                                                                      
[cols="1,1,1", options="header"]                                                
|=================================================                              
| {productname} version | SLES version | Postgresql version                     
| {productname} 3.1.4 | SLES 12 SP3 | Postgres 9.6                                
| {productname} 3.1.11 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 10                                
| {productname} 3.2.10 | SLES 12 SP4 | Postgres 10                                
| {productname} 4.0.0 | SLES 15 SP1 | Postgres 10                                 
| {productname} 4.1.0 | SLES 15 SP2 | Postgres 12                                 
| {productname} 4.2.0 | SLES 15 SP3 | Postgres 13                               
|=================================================                              
                                                                                
                                                                                
////
`

////

[[db-migration-13-prepare]]
== Prepare to Upgrade

Before you begin the upgrade, prepare your existing {productname} Server and create a database backup.

PostgreSQL stores data at [path]``/var/lib/pgsql/data/``.

.Procedure: Preparing to Upgrade

. Check the active PostgreSQL version:
+
----
psql --version
----
+
If you are using PostgreSQL{nbsp}12, you can upgrade to PostgreSQL{nbsp}13.
+
If you are already using PostgreSQL version 13, you do not need to perform this migration.
. Check the active smdba version:
+
----
rpm -q smdba
----
+
// CHECKIT!
+
PostgreSQL{nbsp}13 requires ``smdba`` version 1.7.6 or later.

. Perform a database backup.
    For more information on backing up, see xref:administration:backup-restore.adoc[].

> 

Reviewer 2: Needs some whitespace adjustments, but should render OK.

Author: which whitespaces do you refer to?

Reviewer 2: No need to put newlines between steps, and we usually put three newlines before a title. But it's just cosmetic, so no big deal.

Author: jejeje, what is "steps" and what is "title" ?

Reviewer 2: Let's just merge this as-is, and I'll fix the table in a new PR.

[[db-migration-13-upgrade]]
== Upgrade PostgreSQL

[WARNING]
====
Always create a database backup before performing a migration.
====

PostgreSQL upgrades can be performed in two ways: a regular upgrade, or a fast upgrade:

> A regular upgrade will create a complete copy of the database, so you will need double the existing database size of space available.

Reviewer 2: ```suggestion
A regular upgrade creates a complete copy of the database, so you need double the existing database size of space available.
```
